### DISCUSSION TITLE
Getting Started?

### DISCUSSION THREAD

**Discussion by TjMOTS** [TheTrackerCouncil/SMZ3Randomizer#373] [Q&A]
Hello I've been self teaching, Python, C++, and HTML. I eventually want to get a software dev job and heard that having a good Github commit history looks good during an interview. So I'm curious how to use Github effectively. I would be willing to spend time coding in exchange for some guidance on where to start.

**Comment by Vivelin** [comment]
Hi! I'm sorry for not getting back to you sooner, but I've been away for personal reasons. This probably isn't the right place as we use C# here, but I'll try to answer your question as best I can as I'm also a self-taught dev.

If you have no professional experience and no formal education, the best place to start — in my opinion — would be to start putting stuff you've made up on GitHub. When I first got a job before I had any experience, all I had to show were the hobby projects I had made. It doesn't have to be perfect. And if you don't know much about source control yet, that'll help you gain some experience with it too.

A better place to ask questions like this would probably be over at [StackOverflow] or the [Software Engineering Stack Exchange]. Good luck!

[StackOverflow]: https://stackoverflow.com/
[Software Engineering Stack Exchange]: https://softwareengineering.stackexchange.com/
